### PR TITLE
feat(test-toolkit): Create documentation for test extension and users

### PR DESCRIPTION
**Pull request by alachambre**

### CODE REVIEW

File: modules/process-testing/pages/user.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+== Retrieve a User
+
+Users are mainly used to start processes and execute tasks. Any user in the organization can be retrieved using its username and the Bonita Test Toolkit: 
+
+[source, Java]
+----
+class ProcessIT {
+
+    @RegisterExtension
```
Comment (rbioteau) [inline modules/process-testing/pages/user.adoc]: Prefer the `@BonitaTests` annotation

File: modules/process-testing/pages/bonita-test-extension.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+**- Target runtime URL**: The targeted Bonita runtime URL. It is used to perform HTTP requests to interact with this runtime.
+
+**- Delete process instances**: Whether all existing process instances (including archived instances and their documents) should be deleted before tests execution.
+
+**- Clear BDM**: Whether the BDM database of the targeted Bonita should be cleared before tests execution. All existing business data will be removed.
+
+[source, Java]
+----
+ @RegisterExtension <1>
```
Comment (rbioteau) [inline modules/process-testing/pages/bonita-test-extension.adoc]: Prefer the `@BonitaTests` annotation

You may document both way.

**Review by rbioteau — CHANGES_REQUESTED** [review]

File: modules/process-testing/pages/bonita-test-extension.adoc (hunk 2 above)
Comment (alachambre) [reply modules/process-testing/pages/bonita-test-extension.adoc]: Hum OK. I think I'll also document the way to configure test container in this page.

Comment (rbioteau) [reply modules/process-testing/pages/bonita-test-extension.adoc]: I don't think we should document testcontainers as the goal is to use it with BCD on  a provisionned environment

Comment (alachambre) [reply modules/process-testing/pages/bonita-test-extension.adoc]: As you wish, but in my view the best solution is to use test container and the builder and the deployer. It's sad to see that we do not want to provide our users the best tooling to use.

**Pull request merged**